apps/src/templates/AppView.jsx

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv.jsx');
+
+/**
+ * Top-level React wrapper for our standard blockly apps.
+ */
+var AppView = React.createClass({
+  propTypes: {
+    renderCodeApp: React.PropTypes.func.isRequired,
+    onMount: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount: function () {
+    this.refs.codeApp.getDOMNode().innerHTML = this.props.renderCodeApp();
+    this.props.onMount();
+  },
+
+  render: function () {
+    return <ProtectedStatefulDiv ref="codeApp" />;
+  }
```

Comment:
Awesome component! I don't like that the parent is rendering the child's contents, however this does have the benefit of ensuring that the required child component does get rendered. If this is intentional, let's have a comment explaining that we're doing it this way to ensure the required component is rendered.

Alternately, the rendering method could be passed to the `ProtectedStatefulDiv` which would then render itself in its own `componentDidMount` method, and we'd have an assertion or something here to ensure that all required components (or DOM element IDs) exist in the DOM. However this is unreliable because you can forget to assert.

The scenario here is that e.g. we always render code mode (visible) and design mode (hidden), but on levels when we show design mode first we might forget to initially render code mode. This would be nice to catch at render time (just as it would be nice to catch a required component unmounting at the time it is unmounted as `ProtectedStatefulDiv` currently does).

Bear with me for some brainstorming that may nor may not reflect something we want to spend time on now:

In an amazing world, we could use a pattern which knows that component A has a required, protected child component B (or possibly required DOM element id `#b`). I first thought of mixins, but apparently mixins are dead in favor of composition: https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750#.kp8veb9xf Following the example laid out in the article, you could have a function which says "wrap Component A by making sure it (initially and always) has a child DOM node with id `#b`":
```
var AppView = React.createClass({
  ...
  render: function () {
    return <div id="codeApp" />
  }
});
AppView = requiredChildren(AppView, ['codeApp']);
function requiredChildren(Component, children) {
  // Magic trick to make sure child component is never unmounted, this part may be hard
}
```
Or something like that. To reiterate, I don't think this is strictly necessary but could be a useful if we think we'll use this pattern a lot.